src/api/become-observed.ts

```diff
@@ -38,16 +38,25 @@ function interceptHook(hook: "onBecomeObserved" | "onBecomeUnobserved", thing, a
     const atom: IObservable =
         typeof arg2 === "string" ? getAtom(thing, arg2) : (getAtom(thing) as any)
     const cb = typeof arg2 === "string" ? arg3 : arg2
-    const orig = atom[hook]
+    const listenersKey = `${hook}Listeners`
+
+    if (!atom[listenersKey]) {
+        atom[listenersKey] = new Set<Lambda>([cb])
+    } else {
+        ;(atom[listenersKey] as Set<Lambda>).add(cb)
```

Comment:
Why you used type assertion instead of editing `IObservable` interface?

Now the type system allows existence of observable implementation which does not know about listeners...